File: javascript/0/01/Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function ( nums, target ) {
  let map = {};
  for ( var i = 0; i < nums.length; i++ ) {
    let curNum = nums[i];
    if ( map[curNum] > -1 ) {
      return [i, map[curNum]];
    }
    map[target - curNum] = i;
  }
};

File: csharp/0/01/Two Sum.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LeetCodeTraining.CSharp._0._01 {
    class Two_Sum {
        public int[] TwoSum(int[] nums, int target) {
            var map = new Dictionary<int, int>();
            for (int i = 0; i < nums.Length; i++) {
                var curNum = nums[i];
                if (map.TryGetValue(curNum, out int matchIndex)) {
                    return new int[] { i, matchIndex };
                }
                if (map.ContainsKey(target - curNum)) {
                    continue;
                }
                map.Add(target - curNum, i);
            }
            return new int[] { 0, 0 };
        }
    }
}
